test/intl402/NumberFormat/prototype/format/unit-de-DE.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.numberformat.prototype.format
+description: Checks handling of the unit style.
+locale: [de-DE]
+features: [Intl.NumberFormat-unified]
+---*/
+
+
+const tests = [
+  [
+    -987,
+    {
+      "short": "-987 m",
+      "narrow": "-987 m",
+      "long": "-987 Meter",
```

Comment:
This test might be too sensitive to locale data.  @spectranaut @leobalter, thoughts?  If this is OK, then we don't need further changes.  If we need to change this to be less sensitive to locale data, we could do something along the lines of:

1. Verify that `formatToParts` output, when concatenated, equals `format` output (actually, this is probably something all the `formatToParts` tests should do anyhow)
2. Verify that there is a nonempty entry in the `formatToParts` output with `type: "unit"`

Reply:
I believe the data is just fine, considering this is a specific test for a specific locale. At one point there is some mitigation over having tests for the same units using different locales, therefore it's not required - although not forbidden - to have multiple.

At the same time, the idea for an almost generic test using formatToParts would be a nice addition, like in using a default locale. I'm +1 to these two items without necessarily removing the current items.